botocove/cove_host_account.py

```diff
@@ -102,28 +103,107 @@ def _get_boto3_sts_client(self, assuming_session: Optional[Session]) -> STSClien
 
     def _resolve_target_accounts(self, target_ids: Optional[List[str]]) -> Set[str]:
         if self.provided_ignore_ids:
-            validated_ignore_ids = self._format_ignore_ids()
+            ignore_accounts, ignore_ous = self._get_validated_ids(
+                self.provided_ignore_ids, "ignore_ids"
+            )
+            ignore_accounts_from_ous = self._get_all_accounts_by_organization_units(
+                ignore_ous
+            )
+            ignore_accounts.extend(ignore_accounts_from_ous)
+            ignore_accounts_set = set(ignore_accounts)
         else:
-            validated_ignore_ids = set()
+            ignore_accounts_set = set()
 
         if target_ids is None:
             # No target_ids passed
-            target_accounts = self._gather_org_assume_targets()
+            target_accounts_set = self._gather_org_assume_targets()
         else:
-            # Specific list of IDs passed
-            target_accounts = set(target_ids)
-
-        return target_accounts - validated_ignore_ids
-
-    def _format_ignore_ids(self) -> Set[str]:
-        if not isinstance(self.provided_ignore_ids, list):
-            raise TypeError("ignore_ids must be a list of account IDs")
-        for account_id in self.provided_ignore_ids:
-            if len(account_id) != 12:
-                raise TypeError("All ignore_id in list must be 12 character strings")
-            if not isinstance(account_id, str):
-                raise TypeError("All ignore_id list entries must be strings")
-        return set(self.provided_ignore_ids)
+            target_accounts, target_ous = self._get_validated_ids(
+                target_ids, "target_ids"
+            )
+            target_accounts_from_ous = self._get_all_accounts_by_organization_units(
+                target_ous
+            )
+            target_accounts.extend(target_accounts_from_ous)
+            target_accounts_set = set(target_accounts)
+
+        return target_accounts_set - ignore_accounts_set
+
+    def _get_validated_ids(
+        self, ids: List[str], ids_type: str
```

Comment:
`ids_type` doesn't seem worth the complexity of an arg - the user can guess what they've mangled - I don't mind putting the burden of knowing what an OU or acc ID should look like on the user if they're using that feature!